### PR TITLE
Rewrite bytecode with a sourcemap

### PR DESCRIPTION
**Pull request by swankjesse**

### CODE REVIEW

File: buildSrc/src/main/kotlin/Dependencies.kt [hunk 1]
```diff
@@ -23,6 +23,7 @@ object Dependencies {
   val kotlinGradlePlugin = "org.jetbrains.kotlin:kotlin-gradle-plugin:${baseKotlin}"
   val kotlinReflect = "org.jetbrains.kotlin:kotlin-reflect"
   val kotlinSerialization = "org.jetbrains.kotlin:kotlin-serialization:${baseKotlin}"
+  val kotlinTest = "org.jetbrains.kotlin:kotlin-test:1.5.31"
```
Comment (JakeWharton) [inline buildSrc/src/main/kotlin/Dependencies.kt]:
```suggestion
  val kotlinTest = "org.jetbrains.kotlin:kotlin-test"
```
You'll get the right version automatically

File: zipline-bytecode/src/test/kotlin/app/cash/zipline/bytecode/AtomSetTest.kt [hunk 2]
```diff
@@ -0,0 +1,50 @@
...
+
+import com.google.common.truth.Truth.assertThat
+import kotlin.test.assertFailsWith
+import org.junit.Test
+
+class AtomSetTest {
+  @Test fun happyPath() {
+    val atomSet = MutableAtomSet(listOf("hello", "world"))
+
+    // Null and true are likely to be atoms.
```
Comment (JakeWharton) [inline zipline-bytecode/src/test/kotlin/app/cash/zipline/bytecode/AtomSetTest.kt]:
```suggestion
    // Null and true are built-in atoms.
```

File: zipline-bytecode/src/test/kotlin/app/cash/zipline/bytecode/AtomSetTest.kt [hunk 3]
```diff
@@ -0,0 +1,50 @@
...
+    assertThat(atomSet.idOf("null")).isAtLeast(0)
+    assertThat(atomSet.idOf("true")).isAtLeast(0)
+
+    // Null and true are likely to be atoms.
```
Comment (JakeWharton) [inline zipline-bytecode/src/test/kotlin/app/cash/zipline/bytecode/AtomSetTest.kt]:
```suggestion
```

**Review by JakeWharton — APPROVED** [review]

File: zipline-bytecode/src/main/kotlin/app/cash/zipline/bytecode/applySourceMap.kt [hunk 4]
```diff
@@ -0,0 +1,101 @@
...
+ *  See the License for the specific language governing permissions and
+ *  limitations under the License.
+ */
+package app.cash.zipline.bytecode
+
+import okio.Buffer
+
+/**
+ * Combines QuickJS bytecode with a JavaScript source map to produce new QuickJS bytecode that
+ * bakes in the source map. Exceptions thrown will have the stack traces aa if the original sources
```
Comment (Egorand) [inline zipline-bytecode/src/main/kotlin/app/cash/zipline/bytecode/applySourceMap.kt]:
```suggestion
 * bakes in the source map. Exceptions thrown will have the stack traces as if the original sources
```

**Review by Egorand — APPROVED** [review]

File: buildSrc/src/main/kotlin/Dependencies.kt (hunk 1 above)
Comment (swankjesse) [reply buildSrc/src/main/kotlin/Dependencies.kt]: Fixed.

**Pull request merged**